### PR TITLE
Show description dispute reasons in order notes

### LINKED ISSUE #6474: Ensure consistent naming of dispute types in WCPay client
### Describe the bug

When an order is disputed in WCPay, an order note is generated showing the reason for the dispute. The reason does not match the descriptions given in emails (server) and in the Dispute details.

### To Reproduce

1. Purchase a product using a ["dispute" test card](https://stripe.com/docs/testing#disputes) `4000000000000259` will automatically create a dispute with a reason of `fraudulent`
2. Viewing the dispute details under `Payments > Disputes`, will show `Transaction unauthorized` with #6472.
3. View the relevant order.
4. Order note will show the reason is `fraudulent`

### Screenshots
image

### Expected behavior

The reasons given for the dispute should match across the entire WCPay product.

### Additional context

Relates to #6472 and #6381

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #6474 

#### Changes proposed in this Pull Request

When an order is disputed in WCPay, an order note is generated showing the reason for the dispute. Previously, this reason did not match the descriptions given in emails (server) and in the Dispute details.

**Before**
image

**After**
image

#### Testing instructions

1. Purchase a product using a ["dispute" test card](https://stripe.com/docs/testing#disputes) `4000000000000259` will automatically create a dispute with a reason of `fraudulent`
2. Viewing the dispute details under `Payments > Disputes`, will show `Transaction unauthorized` with #6472.
3. View the relevant order.
5. Order note will show the dispute reason is `Transaction unauthorized`

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6475` or branch name `fix/6474-consistent-dispute-naming` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6474-consistent-dispute-naming) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: cfa0977c2135f5184a9f39c614edc87c238f13a9
- Build time: 2023-06-20 22:19:45 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-utils.php [hunk 1]
```diff
@@ -887,4 +887,47 @@ public static function get_wpcore_request_class(): string {
...
+	 * This mapping is duplicated in client/disputes/strings.ts and on Server.
+	 *
+	 * @param string $reason The dispute reason.
+	 *
+	 * @return string
+	 */
+	public static function get_dispute_reason_description( string $reason ): string {
+		switch ( $reason ) {
+			case 'bank_cannot_process':
+				return __( 'Bank can not process', 'woocommerce-payments' );
```
Comment (Jinksi) [inline includes/class-wc-payments-utils.php]: I think this should be `Bank cannot process` which is consistent with `client/disputes/strings.ts` and the changes you've proposed in https://github.com/Automattic/woocommerce-payments-server/pull/3436.

https://github.com/Automattic/woocommerce-payments/blob/7cef6df74173a4d6e278f08ec4346b588e48bb5d/client/disputes/strings.ts#L22

```suggestion
				return __( 'Bank cannot process', 'woocommerce-payments' );
```

File: includes/class-wc-payments-utils.php [hunk 2]
```diff
@@ -887,4 +887,47 @@ public static function get_wpcore_request_class(): string {
...
+			case 'incorrect_account_details':
+				return __( 'Incorrect account details', 'woocommerce-payments' );
+			case 'insufficient_funds':
+				return __( 'Insufficient funds', 'woocommerce-payments' );
+			case 'product_not_received':
+				return __( 'Product not received', 'woocommerce-payments' );
+			case 'product_unacceptable':
+				return __( 'Product unacceptable', 'woocommerce-payments' );
+			case 'subscription_canceled':
+				return __( 'Subscription cancelled', 'woocommerce-payments' );
```
Comment (Jinksi) [inline includes/class-wc-payments-utils.php]: As above, I think this should be `Subscription canceled` which is consistent with `client/disputes/strings.ts` and the changes you've proposed in https://github.com/Automattic/woocommerce-payments-server/pull/3436.

https://github.com/Automattic/woocommerce-payments/blob/7cef6df74173a4d6e278f08ec4346b588e48bb5d/client/disputes/strings.ts#L254


```suggestion
				return __( 'Subscription canceled', 'woocommerce-payments' );
```

File: changelog/fix-6474-consistent-dispute-naming [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Dispute reasons inconsistencies have been fixed.
```
Comment (Jinksi) [inline changelog/fix-6474-consistent-dispute-naming]: Should we mention that this includes a change to order notes, perhaps as an additional changelog entry?
```suggestion
Show descriptive dispute reasons in order notes.
```

File: includes/class-wc-payments-utils.php [hunk 4]
```diff
@@ -887,4 +887,47 @@ public static function get_wpcore_request_class(): string {
...
+			case 'credit_not_processed':
+				return __( 'Credit not processed', 'woocommerce-payments' );
+			case 'customer_initiated':
+				return __( 'Customer initiated', 'woocommerce-payments' );
+			case 'debit_not_authorized':
+				return __( 'Debit not authorized', 'woocommerce-payments' );
+			case 'duplicate':
+				return __( 'Duplicate', 'woocommerce-payments' );
+			case 'fraudulent':
+				return __( 'Transaction unauthorized', 'woocommerce-payments' );
```
Comment (Jinksi) [inline includes/class-wc-payments-utils.php]: To keep this PR changelog correct, should this be `Fraudulent` to remain consistent with `client/disputes/strings.ts`?

https://github.com/Automattic/woocommerce-payments/blob/7cef6df74173a4d6e278f08ec4346b588e48bb5d/client/disputes/strings.ts#L112

I see that this is being changed to `Transaction unauthorized` in #6472, so perhaps that PR could handle the change across both files.

```suggestion
				return __( 'Fraudulent', 'woocommerce-payments' );
```

**Review by Jinksi — CHANGES_REQUESTED** [review]
Thanks @brucealdridge, looking good 👍

I've requested changes to address some inconsistencies that remain between PHP & JS dispute reasons.

File: includes/class-wc-payments-utils.php (hunk 4 above)
Comment (brucealdridge) [reply includes/class-wc-payments-utils.php]: Good thinking, I'll update the other PR

**Comment by brucealdridge** [comment]
> Thanks @brucealdridge, looking good 👍
> 
> I've requested changes to address some inconsistencies that remain between PHP & JS dispute reasons.

😃 Good spotting, I'm not sure how I missed those changes given the title of the PR. I think I was focused on the order notes change!

**Review by Jinksi — APPROVED** [review]
LGTM @brucealdridge 🙌 

I've tested this locally using the fraudulent dispute test card number and see the "Fraudulent" display reason rendered in the order note.

image

**Pull request merged**